### PR TITLE
Use configurable KnownAuthorityHosts for smoke tests (includes dynamic resource provisioning)

### PR DESCRIPTION
**Pull request by danieljurek**
Maps the environment variable `AZURE_AUTHORITY_HOST_ALIAS` to the various `KnownAuthorityHosts` when using `DefaultAzureCredential` 

This also has support for the China Cloud and dynamic resource provisioning built in. The two efforts happened to dovetail.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: common/SmokeTests/SmokeTest/KeyVaultTest.cs [hunk 1]
```diff
@@ -30,17 +37,12 @@ public static async Task RunTests()
...
             Console.WriteLine("3.- Delete that Secret (Clean up)\n");
 
-            string tenantID = Environment.GetEnvironmentVariable("DIR_TENANT_ID");
-            string clientID = Environment.GetEnvironmentVariable("APP_CLIENT_ID");
-            string clientSecret = Environment.GetEnvironmentVariable("CLIENT_SECRET");
             string keyVaultUri = Environment.GetEnvironmentVariable("KEY_VAULT_URI");
-            
-            var authorityHost = Environment.GetEnvironmentVariable("AZURE_AUTHORITY_HOST") 
-                ?? defaultAuthorityHost;
+            var authorityHost = GetAuthorityHost(Environment.GetEnvironmentVariable("AZURE_AUTHORITY_HOST_ALIAS"), KnownAuthorityHosts.AzureCloud);
```
Comment (danieljurek) [inline common/SmokeTests/SmokeTest/KeyVaultTest.cs]: Use `AZURE_CLOUD` instead (from @jongio)

File: common/SmokeTests/SmokeTest/test-resources.json [hunk 2]
```diff
@@ -282,7 +282,7 @@
             "type": "string",
             "value": "[parameters('testApplicationSecret')]"
         },
-        "AZURE_AUTHORITY_HOST_ALIAS": {
+        "AZURE_CLOUD": {
             "type": "string",
             "value": "[parameters('authorityHostAlias')]"
```
Comment (jongio) [inline common/SmokeTests/SmokeTest/test-resources.json]: Would be good to update the param name too.

**Review by jongio — CHANGES_REQUESTED** [review]

File: eng/pipelines/smoke-tests.yml [hunk 3]
```diff
@@ -59,9 +59,9 @@ jobs:
       vmImage: $(OSVmImage)
 
     variables:
-      azureCloudArmParameters: "@{ storageEndpointSuffix = 'core.windows.net'; authorityHost = 'https://login.microsoftonline.com'; }"
-      azureUSGovernmentArmParameters: "@{ storageEndpointSuffix = 'core.usgovcloudapi.net'; authorityHost = 'https://login.microsoftonline.us/'; }"
-      azureChinaCloudArmParameters: "@{ storageEndpointSuffix = 'core.chinacloudapi.cn'; authorityHost = 'https://login.chinacloudapi.cn'; }"
+      azureCloudArmParameters: "@{ storageEndpointSuffix = 'core.windows.net'; authorityHostAlias = 'AzureCloud'; }"
```
Comment (jongio) [inline eng/pipelines/smoke-tests.yml]: Same here - Update param name?

**Review by jongio — APPROVED** [review]

**Pull request merged**